scipy/stats/_hypotests.py

```diff
@@ -144,6 +144,219 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     return Epps_Singleton_2sampResult(w, p)
 
 
+PoissonMeansTestResult = make_dataclass('PoissonMeansTestResult',
+                                        ('statistic', 'pvalue'))
+
+
+def poisson_means_test(k1, n1, k2, n2, diff=0, alternative='two-sided'):
+    r"""
+    Calculates the poisson means test, the "E-test", for the mean difference of
+    two samples that follow a Poisson distribution from descriptive statistics.
+
+    This is a two-sided test. The null hypothesis is that two independent
+    samples have identical average (expected) values.
+
+    Let :math:`X_{11},...,X_{1n_1}` and :math:`X_{21},...,X_{2n_2}` be
```

Comment:
Let's move this to the notes. Many object to including rendered math anywhere else.
<img width="793" alt="image" src="https://user-images.githubusercontent.com/6570539/162602615-d288fd61-dc42-460d-8920-c1439af2e9b9.png">